plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/server/PodDeploymentPathProvider.java

```diff
@@ -51,18 +51,22 @@
 
 	private static final String DOCKER_IMAGE_DIGEST_IDENTIFIER = "sha256:";
 
-	public String load(IService service, Connection connection) throws CoreException {
-		IProject project = service.getProject();
-		IPod pod = getPod(service, project, connection);
+	public String load(IResource resource, Connection connection) throws CoreException {
+		IProject project = resource.getProject();
+		IPod pod = getPod(resource, project, connection);
 		String imageRef = getImageRef(connection, project, pod);
 		int imageDigestIndex = imageRef.indexOf(DOCKER_IMAGE_DIGEST_IDENTIFIER);
 		if (imageDigestIndex > 0) {
 			String imageDigest = imageRef.substring(imageDigestIndex);
 			IResource imageStreamTag = getImageStreamTag(connection, project, imageRef, imageDigest);
 			return getPodPath(imageStreamTag);
 		} else {
-			
-			return "";
+		    String imageMetaData = getImageMetaData(imageRef, project);
+		    if (imageMetaData != null) {
```

Comment:
this if/else block seems weird:
#getPodPath returns null if the given imageMetaData is null. So this if/else blocks seems to effectively change the semantics by return "" instead. But this isnt consistent: at line 62 this isnt transformed. The null that is potentially returned by #getPodPath is returned as is. I thus doubt that this if/else is even required. Am I missing something?